File: warmup/java/Is_Fibo.java
import java.io.*;
import java.util.*;

public class Is_Fibo {

    public static void main(String[] args) {

        Scanner sc = new Scanner(System.in);

        int T = sc.nextInt();

        long n;

        for(int m=1; m<=T; m++) {
            n = sc.nextLong();

            long fibformula = (5 * Math.pow(n, 2)) + 4;
            long fibFormula2 = (5 * Math.pow(n, 2)) - 4;

            if (isPerfectSquare(fibformula) || isPerfectSquare(fibFormula2)) {
                System.out.println("IsFibo");
            } else {
                System.out.println("IsNotFibo");
            }
        }
    }

    public static boolean isPerfectSquare(long n) {
        double sqrt = Math.sqrt(n);
        if (sqrt % 1 == 0) {
            return true;
        } else {
            return false;
        }
    }
}

File: algorithms/warmup/python2/is-fibo.py
import math


def isperfectsquare(n):
    return True if math.sqrt(n).is_integer() else False

for _ in xrange(int(input())):
    n = int(input())
    fibFormula1 = (5 * (n ** 2)) + 4
    fibFormula2 = (5 * (n ** 2)) - 4
    print "IsFibo" if isperfectsquare(fibFormula1) or isperfectsquare(fibFormula2) else "IsNotFibo"
